site/_shortcodes/Playlist.js

```diff
@@ -0,0 +1,210 @@
+const {html} = require('common-tags');
+const fetch = require('node-fetch/lib/index');
+const apiKey = 'AIzaSyAzOX6K8IXsxPwQ0rgW7QgJPndroPOiWfc';
+const maxResults = 50;
+const part = 'snippet';
+
+/** Renders a a YouTube playlist widget
+ * @param {string} playlistId is a YouTube playlist id
+ */
+async function Playlist(playlistId) {
+  let videoNumber = 1;
+  let videoTotal = 0;
+
+  // Set some empty variables to populate
+  let channelId = '';
+  let channelName = '';
+  let channelThumb = '';
+  let playlistName = '';
+  let playlistFirstVideo = '';
+  let playlistThumb = '';
+  let playlistUpdated = '';
+  let playlistHtml = '';
+
+  /*
+    Playlist Information
+  */
+  await fetch(
+    `https://youtube.googleapis.com/youtube/v3/playlists?part=${part}&id=${playlistId}&key=${apiKey}`
+  )
+    .then(res => res.json())
+    .then(playlistResult => {
+      if (playlistResult.items.length > 0) {
+        channelId = playlistResult.items[0].snippet.channelId;
+        playlistName = playlistResult.items[0].snippet.title;
+        playlistThumb = playlistResult.items[0].snippet.thumbnails.medium.url;
+        playlistUpdated = new Date(
+          playlistResult.items[0].snippet.publishedAt
+        ).toLocaleDateString('en-us', {
+          weekday: 'long',
+          year: 'numeric',
+          month: 'short',
+          day: 'numeric',
+        });
+      } else {
+        return false;
+      }
+
+      // Next Call: Query Channel Information
+      return fetch(
+        `https://youtube.googleapis.com/youtube/v3/channels?part=${part}&id=${channelId}&key=${apiKey}`
+      );
+    })
+    .then(res => res.json())
+    .then(channelResult => {
+      channelName = channelResult.items[0].snippet.title;
+      channelThumb = channelResult.items[0].snippet.thumbnails.medium.url;
+
+      // Next Call: Query Channel Information
+      return fetch(
+        `https://youtube.googleapis.com/youtube/v3/playlistItems?part=${part}&playlistId=${playlistId}&maxResults${maxResults}&key=${apiKey}`
+      );
+    })
+    .then(res => res.json())
+    .then(videosResult => {
+      playlistFirstVideo = videosResult.items[0].snippet.resourceId.videoId;
+      videosResult.items.forEach(video => {
+        playlistHtml += getVideoHtml(video, videoNumber);
+
+        videoNumber++;
+        videoTotal++;
+      });
+    })
+    .catch(error => {
+      console.error(error);
+    });
+
+  return getChannelHtml(
```

Comment:
Add a sensible fallback for render if the API information can not be retrieved, for example if no API key is set.